runners/flink/runner/src/main/java/org/apache/beam/runners/flink/TestFlinkRunner.java

```diff
@@ -79,6 +86,31 @@ public PipelineResult run(Pipeline pipeline) {
   public PipelineOptions getPipelineOptions() {
     return delegate.getPipelineOptions();
   }
+
+  private void assertAssertionCounters(Pipeline pipeline, FlinkRunnerResult result) {
+    int expectedNumberOfAssertions = PAssert.countAsserts(pipeline);
+    Integer succeededAssertions = 0;
+    try {
+      succeededAssertions = result.getAggregatorValue(PAssert.SUCCESS_COUNTER);
+    } catch (AggregatorRetrievalException e) {
+      // No assertions registered will cause an AggregatorRetrievalException here.
+    }
+    Integer failedAssertions = 0;
+    try {
+      failedAssertions = result.getAggregatorValue(PAssert.FAILURE_COUNTER);
+    } catch (AggregatorRetrievalException e) {
+      // No assertions registered will cause an AggregatorRetrievalException here.
+    }
+    assertThat(
```

Comment:
This holds true. All tests that assert this pass. For example: https://github.com/apache/beam/blob/d48c0f3f12920f4d2bdd5e6a719cd9bc8f20cc5f/sdks/java/core/src/test/java/org/apache/beam/sdk/testing/PAssertTest.java#L427-L427

This assertion is a safety in case we have silent failures.